flink-libraries/flink-table/src/main/scala/org/apache/flink/table/calcite/RelTimeIndicatorConverter.scala

```diff
@@ -166,9 +167,20 @@ class RelTimeIndicatorConverter(rexBuilder: RexBuilder) extends RelShuttle {
     val right = join.getRight.accept(this)
 
     LogicalJoin.create(left, right, join.getCondition, join.getVariablesSet, join.getJoinType)
-
   }
 
+  def visit(temporalJoin: LogicalTemporalTableJoin): RelNode = {
+    val left = temporalJoin.getLeft.accept(this)
+    val right = temporalJoin.getRight.accept(this)
+
+    val rewrittenTemporalJoin = temporalJoin.copy(temporalJoin.getTraitSet, List(left, right))
+
+    // Materialize all of the time attributes from the right side of temporal join
+    val indicesToMaterialize =
```

Comment:
The processing time of the right side must not be materialized.

Reply:
As we discussed, it is probably better to materialise it to make it more consistent.